lib/libbackscrub.cc

```diff
@@ -0,0 +1,270 @@
+/* This is licenced software, @see LICENSE file.
+ * Authors - @see AUTHORS file.
+==============================================================================*/
+
+// tested against tensorflow lite v2.4.1 (static library)
+
+#include "tensorflow/lite/interpreter.h"
+#include "tensorflow/lite/kernels/register.h"
+#include "tensorflow/lite/model.h"
+#include "tensorflow/lite/optional_debug_tools.h"
+
+#include "transpose_conv_bias.h"
+#include "libbackscrub.h"
+
+// Debug helper
+static void _dbg(calcinfo_t &info, const char *fmt, ...) {
+	va_list ap;
+	va_start(ap, fmt);
+	if (info.ondebug)
+		info.ondebug(info.caller_ctx, fmt, ap);
+	else
+		vfprintf(stderr, fmt, ap);
+	va_end(ap);
+}
+
+// Tensorflow Lite helper functions
+enum class modeltype_t {
+	Unknown,
+	BodyPix,
+	DeepLab,
+	GoogleMeetSegmentation,
+	MLKitSelfie,
+};
+
+struct normalization_t {
+	float scaling;
+	float offset;
+};
+
+struct backscrub_ctx_t {
+	std::unique_ptr<tflite::FlatBufferModel> model;
+	std::unique_ptr<tflite::Interpreter> interpreter;
+	modeltype_t modeltype;
+	normalization_t norm;
+};
+
+static cv::Mat getTensorMat(calcinfo_t &info, int tnum) {
+
+	backscrub_ctx_t &ctx = *((backscrub_ctx_t *)info.backscrub_ctx);
+	TfLiteType t_type = ctx.interpreter->tensor(tnum)->type;
+	TFLITE_MINIMAL_CHECK(t_type == kTfLiteFloat32);
+
+	TfLiteIntArray* dims = ctx.interpreter->tensor(tnum)->dims;
+	if (info.debug)
```

Comment:
This check should be in `_dbg` …

Reply:
Heh, order of implementation bug, I'll go find any other places this may have occurred too.

Reply:
Removed entirely, debug filtering is the job of library caller if required.